controllers/profiles/getProfileByName.js

```diff
@@ -63,13 +63,20 @@ module.exports = async (req, res) => {
     clonedUser.following_symbol = checkMoreOrLess(getFollowingCountResult);
     clonedUser.follower_symbol = checkMoreOrLess(getFollowerCountResult);
 
-    clonedUser.isSignedMessageEnabled = false;
+    if (req?.userId) {
+      clonedUser.is_following = isFollowingResult;
+    }
+
+    clonedUser.isSignedMessageEnabled = true;
     clonedUser.isAnonMessageEnabled = false;
 
-    clonedUser.is_following = isFollowingResult;
-    if (clonedUser.only_received_dm_from_user_following)
-      clonedUser.isSignedMessageEnabled = isFollowingResult;
-    else clonedUser.isSignedMessageEnabled = true;
+    if (clonedUser.allow_anon_dm) {
+      if (clonedUser.only_received_dm_from_user_following) {
+        clonedUser.isAnonMessageEnabled = clonedUser.allow_anon_dm && isFollowingResult;
+      } else {
+        clonedUser.isAnonMessageEnabled = clonedUser.allow_anon_dm;
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code checks if anonymous messaging is allowed and sets `isAnonMessageEnabled` accordingly. It also considers whether the user only receives direct messages from followers. This seems like a good approach to handle different scenarios for anonymous messaging. However, there's a potential issue here: if `allow_anon_dm` is false, `isAnonMessageEnabled` will remain false regardless of the value of `only_received_dm_from_user_following`. If you want `isAnonMessageEnabled` to be true when `only_received_dm_from_user_following` is true (even if `allow_anon_dm` is false), you should adjust the logic accordingly.

```diff
    if (clonedUser.allow_anon_dm) {
      if (clonedUser.only_received_dm_from_user_following) {
        clonedUser.isAnonMessageEnabled = clonedUser.allow_anon_dm && isFollowingResult;
      } else {
        clonedUser.isAnonMessageEnabled = clonedUser.allow_anon_dm;
      }
    } else if (clonedUser.only_received_dm_from_user_following) {
      clonedUser.isAnonMessageEnabled = isFollowingResult;
    }
```

Please ensure that this proposed change aligns with your application's requirements before implementing it.


<!-- This is an auto-generated comment by OSS CodeRabbit -->